synapse/storage/databases/main/events.py

```diff
@@ -1285,15 +1289,22 @@ def _update_outliers_txn(self, txn, events_and_contexts):
             to_remove.add(event)
 
             if context.rejected:
-                # If the event is rejected then we don't care if the event
-                # was an outlier or not.
+                # If the incoming event is rejected then we don't care if the event
+                # was an outlier or not - what we have is at least as good.
                 continue
 
             outlier_persisted = have_persisted[event.event_id]
             if not event.internal_metadata.is_outlier() and outlier_persisted:
                 # We received a copy of an event that we had already stored as
                 # an outlier in the database. We now have some state at that
                 # so we need to update the state_groups table with that state.
```

Comment:
(I know you didn't edit this sentence, but...) this feels like it is missing a word "We now have state at that so": that *what*? Maybe:

> We now have some state at that event, so we need to update the state_groups table with that state.

Reply:
sure, done.